### PR TITLE
feat(entries): add number field component

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Not used by any property so far, but could be important to have it in the future.

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
Let's wait with merging this when the split up is done (cf. #5)

**Comment by pinussilvestrus** [comment]
Ready to review again.

**Comment by barmac** [comment]
Perhaps we should also add `step` if we want to make it complete. Apart from that it looks good.

**Comment by pinussilvestrus** [comment]
> Perhaps we should also add `step` if we want to make it complete. Apart from that it looks good.

Added 👍 

I updated the reviewers due to the holidays.

File: src/components/entries/NumberField.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  const handleInput = useMemo(() => {
+    return debounce(event => {
+
+      const {
+        validity,
+        value
+      } = event.target;
+
+      if (validity.valid) {
+        onInput(value ? parseInt(value, 10) : undefined);
```
Comment (MaxTru) [inline src/components/entries/NumberField.js:28]: By for example setting `step` to something like `0.01` the user would also be able to insert floats into the number field (e.g., in the example `10.12` would be a valid input). Still the parseInt will cut off the decimal places (so in the example `10` would be passed to `onInput`). Any reason we do this? Alternatively we could use `parseFloat` right?

**Review by MaxTru — CHANGES_REQUESTED** [review]

File: src/components/entries/NumberField.js (hunk 1 above)
Comment (pinussilvestrus) [reply src/components/entries/NumberField.js:28]: Good point, I didn't think about it.

If we use `parseFloat`, would that lead to always commit floating point numbers? Let's try it out.

image

Comment (pinussilvestrus) [reply src/components/entries/NumberField.js:28]: Seems like we're good to go with this, also ensuring we can have floating-point numbers via `step`.

Comment (pinussilvestrus) [reply src/components/entries/NumberField.js]: Fixed via https://github.com/bpmn-io/properties-panel/pull/76/commits/b81eedbfc88be75b7c9d877ac217bc9c36a7d784

**Review by MaxTru — APPROVED** [review]

**Pull request merged**